app/build.gradle

```diff
@@ -202,6 +202,25 @@ android.applicationVariants.all { variant ->
         buildConfigField 'String', 'ADJUST_TOKEN', 'null'
         println("--")
     }
+
+// -------------------------------------------------------------------------------------------------
+// Leanplum: Read token from local file if it exists
+// -------------------------------------------------------------------------------------------------
+
+    print("Leanplum token: ")
+
+    try {
+        def parts = new File("${rootDir}/.leanplum_token").text.trim().split(":")
+        def id = parts[0]
+        def key = parts[1]
+        buildConfigField 'String', 'LEANPLUM_ID', '"' + id + '"'
+        buildConfigField 'String', 'LEANPLUM_TOKEN', '"' + key + '"'
+        println "(Added from .leanplum_token file)"
+    } catch (FileNotFoundException ignored) {
+        buildConfigField 'String', 'LEANPLUM_ID', 'null'
+        buildConfigField 'String', 'LEANPLUM_TOKEN', 'null'
+        println("X_X")
+    }
 }
```

Comment:
Isn't the code missing that actually takes the token (from taskcluster secrets?) and places it in this file?

Reply:
Following up here  #896. My bad I didn't do a great job creating all the sub issues before starting the work. With the move towards breaking tasks up I'm trying to have smaller branches 👍